File: ProPanel/backend/app/api/api_v1/endpoints/metrics.py
import requests
from fastapi import APIRouter, Depends, Query, HTTPException
from app.api import deps
from app.core.config import settings

router = APIRouter()

PROM_URL = settings.PROMETHEUS_URL or "http://localhost:9090"

@router.get("/query")
def prometheus_query(
    q: str = Query(..., description="PromQL запрос"),
    current_user=Depends(deps.get_current_active_user)
):
    try:
        resp = requests.get(f"{PROM_URL}/api/v1/query", params={"query": q})
        data = resp.json()
        return data
    except Exception as e:
        raise HTTPException(500, f"Ошибка запроса к Prometheus: {str(e)}")

@router.get("/alerts")
def prometheus_alerts(current_user=Depends(deps.get_current_active_user)):
    try:
        resp = requests.get(f"{PROM_URL}/api/v1/alerts")
        return resp.json()
    except Exception as e:
        raise HTTPException(500, f"Ошибка запроса к Prometheus: {str(e)}")

File: ProPanel/frontend/src/api/metrics.ts
import axios from "axios";
import { getToken } from "./auth";

export async function queryPrometheus(promql: string) {
  const resp = await axios.get("/api/v1/metrics/query", {
    headers: { Authorization: `Bearer ${getToken()}` },
    params: { q: promql },
  });
  return resp.data.data;
}

export async function getPrometheusAlerts() {
  const resp = await axios.get("/api/v1/metrics/alerts", {
    headers: { Authorization: `Bearer ${getToken()}` },
  });
  return resp.data.data;
}
